### ISSUE TITLE
Allow disabling syntax validation for blocks

### ISSUE THREAD

**Issue by morremeyer** [hashicorp/vscode-terraform#1612]
### Extension Version

v2.28.2

### Problem Statement

The extension can validate the syntax for blocks, but that does not take into account all possible configurations.

From what I can see, this is mainly a concern with **provider** blocks since many providers can be configured with environment variables.

As an example, take this `vault` provider block:

![Screenshot of a vault provider block with the error "Required attribute "address" not specified: An attribute named "address" is required"](https://github.com/hashicorp/vscode-terraform/assets/7683567/7f6a479f-0ad4-454e-822a-8934e243b9c7)

However, we use Terraform Cloud and define the `VAULT_ADDR` environment variable with a [Variable Set](https://developer.hashicorp.com/terraform/tutorials/cloud/cloud-multiple-variable-sets) that is configured for our workspaces.

There is currently no way to configure the extension to either connect to Terraform Cloud to also take these configurations into account **or** to disable syntax validation for these blocks.

### Expected User Experience

I would like for the syntax validation to not flag correct provider configurations as incorrect.

### Proposal

I would like the syntax validation to support either:

- Being disable for specific blocks. This could be as simple from the user perspective as a comment `# terraform: ignore-next-block` comment
- Allow for some sort of configuration for provider blocks that indicates that a configuration value is set from an environment variable. This configuration can then be taken into account by the syntax validation. This could even be a terraform core feature, making configuration easier to understand (since currently, nothing tells users that a certain value is read from the environment)

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by dbanck** [comment]
Thank you for the suggestion! We started looking into this as part of https://github.com/hashicorp/vscode-terraform/issues/1597, but this is a good, more general summary.

**Comment by morremeyer** [comment]
Thank you! I couldn't find #1597 even though I looked for various combinations of keywords. It might be worth closing it in favour of this issue to keep things in one place 🤔

**Comment by radeksimko** [comment]
Thanks for the report, I have opened https://github.com/hashicorp/vscode-terraform/issues/1616 to describe the problem in a more general way and added some context. It is something we will likely have to address soon as it's obviously wrong/misleading.

I also filed https://github.com/hashicorp/terraform-ls/issues/1480 to track the suggested feature with in-line comments.

I will close this issue for now but feel free to subscribe to either of the two.
